chain/indexer/src/block.rs

```diff
@@ -151,116 +151,126 @@ where
-                                    let snapshot = Snapshot::new(
-                                        U256::from(log.block_number),
-                                        U256::from(log.tx_index), // TODO: unused, kept for ABI compatibility of DB
-                                        log.log_index,
-                                    );
-
-                                    let tx_hash = log.tx_hash;
-
-                                    match db_processor
-                                        .on_event(log.address, log.block_number as u32, log.into(), snapshot)
-                                        .await
-                                    {
-                                        Ok(Some(event_type)) => {
-                                            // Pair the event type with the TX hash here
-                                            let significant_event = SignificantChainEvent { tx_hash, event_type };
-                                            info!("indexer got {significant_event}");
-
-                                            if let Err(e) = tx_significant_events.unbounded_send(significant_event) {
-                                                error!("failed to pass a significant chain event further: {e}");
-                                            }
-                                        }
-                                        Ok(None) => {}
-                                        Err(e) => {
-                                            error!("failed to process logs: {e}");
-                                        }
-                                    };
-                                }
+                    let snapshot = Snapshot::new(
+                        U256::from(log.block_number),
+                        U256::from(log.tx_index), // TODO: unused, kept for ABI compatibility of DB
+                        log.log_index,
+                    );
+
+                    let tx_hash = log.tx_hash;
+
+                    db_processor
+                        .on_event(log.address, log.block_number as u32, log.into(), snapshot)
+                        .await
+                        .map(|v| {
+                            v.map(|event_type| {
+                                // Pair the event type with the TX hash here
+                                let significant_event = SignificantChainEvent { tx_hash, event_type };
+                                debug!("indexer got {significant_event}");
+                                significant_event
+                            })
+                        })
+                        .map_err(|e| {
+                            error!("failed to process logs: {e}");
+                            e
+                        })
+                });
+
+            futures::pin_mut!(block_stream);
+            while let Some(event) = block_stream.next().await {
+                if is_synced.load(std::sync::atomic::Ordering::Relaxed) {
```

Comment:
We should add some `debug!` or `trace!` here (before the `is_synced` check), so we can still see what historical events are being processed.